### PR TITLE
Update contracts to Solidity 0.7.0.

### PR DESCRIPTION
**Pull request by chriseth**
Because I was just working in the area, I upgraded the contracts to be Solidity 0.7.0-compatible.

Of course, truffle cannot yet test it so this is not yet mergeable.

Do you have a way to support multiple versions of Solidity in parallel?

### CODE REVIEW

File: contracts/MockContract.sol [hunk 1]
```diff
@@ -325,11 +327,15 @@ contract MockContract is MockInterface {
 		calldataInvocations[keccak256(abi.encodePacked(resetCount, originalMsgData))] += 1;
 	}
 
-	fallback () payable external {
-		bytes4 methodId;
-		assembly {
-			methodId := calldataload(0)
-		}
+	receive() payable external {
```
Comment (chriseth) [inline contracts/MockContract.sol:330]: The changes here are actually unrelated to 0.7.0.

**Comment by rmeissner** [comment]
I don't think we have a way currently to support multiple Solidity versions in parallel 🤔 @chriseth do you by any chance have an example for that? As far as I can see openzepplin handles this via their releases, so no support for multiple version within the same version.

**Comment by chriseth** [comment]
The only way I see is using different branches or subdirectories. In order to keep the update-lag as small as possible, especially tools/libraries have to update quickly, and I don't see this being compatible with feature-based releases in the libraries.

File: contracts/MockContract.sol (hunk 1 above)
Comment (jiexi) [reply contracts/MockContract.sol:330]: this is a 0.6.x warning, right?

Comment (chriseth) [reply contracts/MockContract.sol:330]: Yes

**Comment by rmeissner** [comment]
@fleupold should be update this and get it to merge?

**Comment by fleupold** [comment]
Yes happy to do that. I'm off tomorrow but can get back to it next week.

**Comment by fleupold** [comment]
Superseeded by #25
